oneview/resource_logical_enclosure.go

```diff
@@ -0,0 +1,358 @@
+						"firmware_baseline_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"firmware_update_on": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"force_install_firmware": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"logical_interconnect_update_mode": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"update_firmware_on_unmanaged_interconnect": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"validate_if_li_firmware_update_is_non_disruptive": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"ip_addressing_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"ipv4_range": {
+				Optional: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"dns_servers": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"domain": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"gateway": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"ip_range_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"subnet_mask": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"logical_interconnect_uris": {
+				Optional: true,
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"power_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"scaling_state": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"scopes_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+
+}
+func resourceLogicalEnclosureCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	logicalEnclosure := ov.LogicalEnclosure{
+		Name:                   d.Get("name").(string),
+		EnclosureGroupUri:      utils.NewNstring(d.Get("enclosure_group_uri").(string)),
+		AmbientTemperatureMode: d.Get("ambient_temperature_mode").(string),
+		DeleteFailed:           d.Get("delete_failed").(bool),
+		ScopesUri:              utils.NewNstring(d.Get("scopes_uri").(string)),
+		Type:                   d.Get("type").(string),
+		Status:                 d.Get("status").(string),
+		URI:                    utils.NewNstring(d.Get("uri").(string)),
+	}
+	enclosureSetCount := d.Get("enclosure_uris.#").(int)
```

Comment:
You should rewrite this to match how you're reading TypeSets below. 
```suggestion
	enclosureSetCount := d.Get("enclosure_uris").(*schema.Set).List()
```

Reply:
We need proper ordering of enclosure uri values, the assignment was wrong when I used TypeSets and creation was failing. Hence converted to TypeList and added ordering.

Reply:
I'm not sure I understand what you mean by the assignment was wrong with TypeSets. Do you mind explaining? Because the API returns an unordered list, we should use TypeSet since it's unordered

Reply:
if you are not ordering the creation is failing . The mapping is coming like the following 

enclosure_uris.2538990670: "" => "/rest/enclosures/0000000000A66101"
enclosure_uris.2774765260: "" => "/rest/enclosures/0000000000A66103 enclosure_uris.3162029965: "" => "/rest/enclosures/0000000000A66102"

If you are ordering then there is no issue with create .

      enclosure_uris.#:    "3"
      enclosure_uris.0:    "/rest/enclosures/0000000000A66101"
      enclosure_uris.1:    "/rest/enclosures/0000000000A66102"
      enclosure_uris.2:    "/rest/enclosures/0000000000A66103"

The above results are after changing to TypeSet and without ordering. The second result is after making it  in order.